### ISSUE TITLE
Incomplete capture when write blocker is used

### ISSUE THREAD

**Issue by bitsgalore** [KBNLresearch/ipmlab#26]
Imaging a Double Density floppy with ddrescue, Windows, using write blocker:

```
ddrescue.exe -b 512 -r4 -v /dev/sdb1 whatever.img whatever.map
```

Result:

```
2022-09-20 16:39:46,167 - INFO - pct rescued:   48.88%, read errors:     7372,  remaining time:         n/a
2022-09-20 16:39:46,167 - INFO -                               time since last successful read:          3s
```

This happens with every DD disk (but not with any HD disks!)

See also <https://github.com/KBNLresearch/ipmlab/issues/26> (duplicate of this issue)

**Comment by bitsgalore** [comment]
Some additional tests:

## Repeat imaging for same floppy without write blocker

Result: works fine!

## Re-try with write blocker

First switched off/on, then inserted same floppy.

Result: works fine.

## Remove DD floppy and insert HD floppy

Result:

```
Press Ctrl-C to interrupt
     ipos:   720896 B, non-trimmed:        0 B,  current rate:   16384 B/s
     opos:   720896 B, non-scraped:        0 B,  average rate:   28356 B/s
non-tried:        0 B,  bad-sector:        0 B,    error rate:       0 B/s
  rescued:   737280 B,   bad areas:        0,        run time:         26s
pct rescued:  100.00%, read errors:        0,  remaining time:         n/a
                              time since last successful read:          0s
```

But this is odd, since HD floppy image size is about 1.5 MB, not 732 kB!

## Retry same HD floppy after resetting write blocker

Result:

```
     ipos:   786432 B, non-trimmed:        0 B,  current rate:       0 B/s
     opos:   786432 B, non-scraped:        0 B,  average rate:   23130 B/s
non-tried:        0 B,  bad-sector:   688128 B,    error rate:   3932 kB/s
  rescued:   786432 B,   bad areas:        1,        run time:         33s
pct rescued:   53.33%, read errors:     6731,  remaining time:         n/a
                              time since last successful read:          5s
```
Note that image is now slightly larger than previous read (which reported no errors!), but only 53% rescued!

## Retry same floppy without write blocker

Result (after interrupting):

```
Press Ctrl-C to interrupt
     ipos:   973312 B, non-trimmed:        0 B,  current rate:       0 B/s
     opos:   973312 B, non-scraped:        0 B,  average rate:    2718 B/s
non-tried:        0 B,  bad-sector:   129024 B,    error rate:       0 B/s
  rescued:    1345 kB,   bad areas:        2,        run time:      8m 14s
pct rescued:   91.25%, read errors:      521,  remaining time:         n/a
                              time since last successful read:      5m 19s
Retrying bad sectors... Retry 2 (backwards)
  Interrupted by user
```
So from these tests it seems that:

1. Using the write blocker can result in incomplete capture because it only exposes part of the file system
2. Read results are generally worse with write blocker.

**Comment by bitsgalore** [comment]
After updating the Tableau USB blocker to the latest firmware, the floppy drive is not recognised anymore at all!

Found this:

<https://resources.culturalheritage.org/emg-review/volume-three-2013-2014/mckinley/>

Quote:

>  PC-formatted 3.5” floppy disks are accessed via a USB floppy drive and imaged with FTK Imager. Although the Tableau USB Write Blocker is not compatible with the USB floppy drive, 3.5” floppy disks can be easily write-blocked by setting a small tab on the disk itself from read/write to read-only.

So perhaps write blocking just isn't feasible (at least not with the Tableau device)?

**Comment by bitsgalore** [comment]
Update - I got in touch with Tableau (the manufacturer of the write blocker), and they confirmed the device may not work well with floppies. From their response:

> To put it in basic terms, as things are right now, to me it sounds like the T8U
> identifies that this is a floppy, so it tries to talk with it as if it is a floppy, but
> in reality this is a USB to adapter device, and considering we might not
> know its language, the two might not talk to each-other properly.

There's no easy solution for this, and they advised not to use the T8U for floppies at this point.
